workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/ConnectionFlows.cs

```diff
@@ -0,0 +1,276 @@
+        {
+            if (UseDevAuthFlow)
+            {
+                var pit = GetDevelopmentPlayerIdentityToken();
+                var loginTokenDetails = GetDevelopmentLoginTokens(parameters.WorkerType, pit);
+
+                LocatorParameters.PlayerIdentity.PlayerIdentityToken = pit;
+                LocatorParameters.PlayerIdentity.LoginToken = SelectLoginToken(loginTokenDetails);
+            }
+
+            using (var locator = new AlphaLocator(LocatorHost, LocatorParameters))
+            {
+                using (var connectionFuture = locator.ConnectAsync(parameters))
+                {
+                    return await Utils.TryToConnectAsync(connectionFuture, token);
+                }
+            }
+        }
+
+        protected virtual string GetDevelopmentPlayerIdentityToken()
+        {
+            var result = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
+                LocatorHost,
+                AnonymousAuthPort,
+                new PlayerIdentityTokenRequest
+                {
+                    DevelopmentAuthenticationToken = DevAuthToken,
+                    PlayerId = GetPlayerId(),
+                    DisplayName = GetDisplayName(),
+                }
+            ).Get();
+
+            if (!result.HasValue)
+            {
+                throw new AuthenticationFailedException("Did not receive a player identity token.");
+            }
+
+            if (result.Value.Status.Code != ConnectionStatusCode.Success)
+            {
+                throw new AuthenticationFailedException("Failed to retrieve a player identity token.\n" +
+                    $"error code: {result.Value.Status.Code}\nerror message: {result.Value.Status.Detail}");
+            }
+
+            return result.Value.PlayerIdentityToken;
+        }
+
+        /// <summary>
+        ///     Retrieves the login tokens for all active deployments that the player
+        ///     can connect to via the anonymous authentication flow.
+        /// </summary>
+        /// <param name="workerType">The type of the worker that wants to connect.</param>
+        /// <param name="playerIdentityToken">The player identity token of the player that wants to connect.</param>
+        /// <returns>A list of all available login tokens and their deployments.</returns>
+        protected virtual List<LoginTokenDetails> GetDevelopmentLoginTokens(string workerType, string playerIdentityToken)
+        {
+            var result = DevelopmentAuthentication.CreateDevelopmentLoginTokensAsync(
+                RuntimeConfigDefaults.LocatorHost,
+                RuntimeConfigDefaults.AnonymousAuthenticationPort,
+                new LoginTokensRequest
+                {
+                    WorkerType = workerType,
+                    PlayerIdentityToken = playerIdentityToken,
+                    UseInsecureConnection = false,
+                    DurationSeconds = 120,
```

Comment:
we may be able to remove this. Think I added it because of a bug in an older version of the worker sdk

Reply:
Leaving as is until we can confirm so